main.py

```diff
@@ -1,29 +1,99 @@
-import sys
-import os
-from file_path_matcher_with_date import FilePathMatcherWithDate
+from constants import CURRENT_DIRECTORY
+
+from validations import (check_directory_path_exists,
+                         validate_year_month,
+                         validate_year
+                         )
+from report_data_provider import ReportDataProvider
+from report_generator import ReportGenerator
+
+import argparse
 
-CURRENT_DIRECTORY = os.getcwd()
 
 if __name__ == "__main__":
 
-    if len(sys.argv) < 4:
-        print("You must specify files path and the command with proper flag")
-        sys.exit(0)
+    arg_parser = argparse.ArgumentParser(
+        description="Generates reports from the weather data files"
+    )
+
+    arg_parser.add_argument(
+        "data_folder",
+        type=str,
+        action='store',
+        nargs=1,
+        metavar="path/to/folder",
+        help="Folder path where data files are stored"
+    )
+
+    arg_parser.add_argument(
+        '-e',
+        type=validate_year,
+        metavar="year",
+        help="To display the highest, lowest "
+             "temperature and highest humidity"
+    )
+
+    arg_parser.add_argument(
+        '-a',
+        type=validate_year_month,
+        metavar="year/mm",
+        help="To display the average high, low "
+             "temperature and average mean humidity"
+    )
+
+    arg_parser.add_argument(
+        '-c',
```

Comment:
the name should be meaningful, it should show the intent. `c, cs, e, a` these should be changed.

Reply:
Yes, but these are arguments given from command line.  Mostly cmd arguments consists of one letter.
like : `python3 main.py -c 2012/6`. In linux, we also give arguments in one letter `ls -a`.